pkg/kubelet/kuberuntime/kuberuntime_gc.go

```diff
@@ -279,31 +298,26 @@ func (cgc *containerGC) evictSandboxes() error {
 			}
 		}
 		if hasContainers {
-			continue
+			sandboxInfo.active = true
 		}
 
-		podUID := types.UID(sandbox.Metadata.Uid)
-		evictSandboxes[podUID] = append(evictSandboxes[podUID], sandboxGCInfo{
-			id:         sandbox.Id,
-			createTime: time.Unix(0, sandbox.CreatedAt),
-		})
+		sandboxesByPod[podUID] = append(sandboxesByPod[podUID], sandboxInfo)
 	}
 
 	// Sort the sandboxes by age.
-	for uid := range evictSandboxes {
-		sort.Sort(sandboxByCreated(evictSandboxes[uid]))
+	for uid := range sandboxesByPod {
+		sort.Sort(sandboxByCreated(sandboxesByPod[uid]))
 	}
 
-	for podUID, sandboxes := range evictSandboxes {
+	for podUID, sandboxes := range sandboxesByPod {
 		if cgc.isPodDeleted(podUID) {
-			// Remove all sandboxes if pod has been deleted.
+			// Remove all evictable sandboxes if the pod has been removed.
```

Comment:
Why not keeping the logic the same with container? I didn't see difference except that sandbox with containers is not evictable.

We could still generate an evictable sandboxes list, and use it here.